analyzer/quality-analyzer/src/main/java/eu/fasten/analyzer/qualityanalyzer/QualityAnalyzerPlugin.java

```diff
@@ -0,0 +1,173 @@
+
+import org.jooq.DSLContext;
+
+
+public class QualityAnalyzerPlugin extends Plugin {
+
+    public QualityAnalyzerPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Extension
+    public static class QualityAnalyzer implements KafkaPlugin, DBConnector {
+
+        private String consumerTopic = "fasten.RapidPlugin.out";
+        private static DSLContext dslContext;
+        private final Logger logger = LoggerFactory.getLogger(QualityAnalyzer.class.getName());
+        private Throwable pluginError = null;
+        private String product = null;
+        private String artifact = null;
+        private String group = null;
+        private String version = null;
+        private JSONObject metrics = null;
+        private boolean restartTransaction = false;
+        private final int transactionRestartLimit = 3;
+        private boolean processedRecord = false;
+
+        @Override
+        public void setDBConnection(DSLContext dslContext) {
+            QualityAnalyzer.dslContext = dslContext;
+        }
+
+        @Override
+        public Optional<List<String>> consumeTopic() {
+            return Optional.of(Collections.singletonList(consumerTopic));
+        }
+
+        @Override
+        public void setTopic(String topicName) {
+            this.consumerTopic = topicName;
+        }
+
+        @Override
+        public void consume(String record) {
+            pluginError = null;
+            product = null;
+            group  =null;
+            artifact = null;
+            version = null;
+            metrics = null;
+            this.processedRecord = false;
+            this.restartTransaction = false;
+            logger.info("Consumed: " + record);
+            var jsonRecord = new JSONObject(record);
+            var payload = new JSONObject();
+            if (jsonRecord.has("payload")) {
+                payload = jsonRecord.getJSONObject("payload");
+            }
+            group = payload.getString("groupId").replaceAll("[\\n\\t ]", "");
+            artifact = payload.getString("artifactId").replaceAll("[\\n\\t ]", "");
+            version = payload.getString("version").replaceAll("[\\n\\t ]", "");
+            metrics = payload.getJSONObject("metrics");
+            int transactionRestartCount = 0;
+            do {
+                try {
+                    var metadataDao = new MetadataDao(dslContext);
+                    dslContext.transaction(transaction -> {
+                        metadataDao.setContext(DSL.using(transaction));
+                        long id;
+                        try {
+                            id = saveToDatabase(group + "." + artifact,
+                                    version,
+                                    metrics,
+                                    metadataDao);
+                        } catch (RuntimeException e) {
+                            logger.error("Error saving data to the database: '" + product + ":" + version + "'", e);
```

Comment:
`":"` is also one of the Constants. Please reference it as `Constants.mvnCoordinateSeparator` everywhere instead of hard-coding `":"`